gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Cluster.java

```diff
@@ -244,9 +241,15 @@ public static Cluster open(final Configuration conf) {
      * Create a {@code Cluster} using a YAML-based configuration file.
      */
     public static Cluster open(final String configurationFile) throws Exception {
-        final File file = new File(configurationFile);
-        if (!file.exists())
-            throw new IllegalArgumentException(String.format("Configuration file at %s does not exist", configurationFile));
+        File file = new File(configurationFile);
+        if (!file.exists()) {
+            ClassLoader currentClassLoader = Thread.currentThread().getContextClassLoader();
```

Comment:
Note that our code style is to `final` variables where possible...could you please add those where necessary?